File: Architecture/重要类定义/Blog.java

import java.time.LocalDateTime;

/**
 * 博文类
 */
public class Blog {
    private String bloguid;     // 博文uid -> 索引
    private String title;       // 博文标题
    private String content;     // 博文内容,存储Markdown源码
    private String summary;     // 摘要
    private Long authorId;      // 作者ID
    private BlogCategory blogCategory;  // 博文分类
    private LocalDateTime createTime;   // 创建日期
    private LocalDateTime updateTime;   // 修改日期
}

File: frontend/web/src/types/blog.ts
import type { ApiId, PageQuery } from './api'
import type { TagBriefVO } from './tag'
import type { UserBriefVO } from './user'

export type PublishStatus = 0 | 1

export interface BlogBaseVO {
    id: ApiId
    title: string
    summary: string | null
    isPublished: PublishStatus
    likeCount: number
    createTime: string
    updateTime: string
    author: UserBriefVO
    tags: TagBriefVO[]
}

export interface BlogPublicBriefVO extends BlogBaseVO {}

export interface BlogPublicDetailVO extends BlogPublicBriefVO {
    content: string | null
    isLiked: boolean
}

export interface BlogPrivateBriefVO extends BlogBaseVO {
    folderId: ApiId
}

export interface BlogPrivateDetailVO extends BlogPrivateBriefVO {
    content: string | null
}

export interface BlogPageQuery extends PageQuery {
    keyword?: string
    authorId?: ApiId
    tagIds?: ApiId[]
}

export interface BlogSaveDTO {
    title: string
    folderId?: ApiId
}

export interface BlogEditDTO {
    folderId: ApiId
    isPublished: PublishStatus
    title: string
    summary?: string
    content?: string
    tagIds?: ApiId[]
}
